File: OOB class/proj1_cpp/owca.cpp
#include "owca.h"
owca::owca(short x, short y, Swiat* swiat, short wiek) {
	this->wiek = wiek;
	this->nazwa = "Owca";
	this->inicjatywa = 4;
	this->sila = 4;
	this->znak = 'o';
	this->polozenie[OX] = x;
	this->polozenie[OY] = y;
	this->swiat = swiat;
	swiat->setJesliWolnePole(x, y, znak);
}

owca::owca(short x, short y, Swiat* swiat, short wiek, short sila, short xp, short yp, short cp) {
	this->wiek = wiek;
	this->nazwa = "Owca";
	this->inicjatywa = 4;
	this->sila = sila;
	this->znak = 'o';
	this->polozenie[OX] = x;
	this->polozenie[OY] = y;
	this->swiat = swiat;
	this->poprzednie_polozenie[OX] = xp;
	this->poprzednie_polozenie[OY] = yp;
	this->cooldown_porodowy = cp;
}

void owca::wykonanie_porodu(short x, short y) {
	swiat->addOrganizm(new owca(x, y, swiat, swiat->get_and_incrementKtoryUrodzony()));
}

File: OOB class/proj2_java/src/symulator/zwierzeta/owca.java
package symulator.zwierzeta;

import symulator.Swiat;
import symulator.Zwierze;

public class owca extends Zwierze {

    public owca(Swiat swiat, int ox, int oy, int wiek, int BOARD_WIDTH, int BOARD_HEIGHT) {
        super(swiat, ox, oy,ox, oy, wiek, BOARD_WIDTH, BOARD_HEIGHT);
        this.nazwa = "Owca";
        this.inicjatywa = 4;
        this.sila = 3;
        this.znak = 'o';
        swiat.setJesliWolnePole(ox, oy, znak);
    }

    @Override
    protected void wykonanie_porodu(int x, int y) {
        swiat.addOrganizm(new owca( swiat,x, y, swiat.get_and_incrementKtoryUrodzony(),this.BOARD_WIDTH,this.BOARD_HEIGHT));
    }



    public owca(Swiat swiat, int x, int y,  int wiek, int sila, int xp, int yp, int cp, int BOARD_WIDTH, int BOARD_HEIGHT) {
        super(swiat, x, y,xp,yp, wiek, BOARD_WIDTH, BOARD_HEIGHT);

        this.nazwa = "Owca";
        this.inicjatywa = 4;
        this.sila = sila;
        this.znak = 'o';
        this.cooldown_porodowy = cp;
    }

}
